destination/format/csv.go

```diff
@@ -24,18 +24,15 @@ import (
 	"golang.org/x/exp/maps"
 )
 
-func MakeCSVBytes(records []sdk.Record, prefix string, orderingColumns []string) (
-	*bytes.Buffer, *bytes.Buffer, map[string]string, []string, []string, error,
-) {
-	var (
-		insertsBuf  bytes.Buffer
-		updatesBuf  bytes.Buffer
-		updateCount int
-		insertCount int
-	)
-
-	insertsWriter := csv.NewWriter(&insertsBuf)
-	updatesWriter := csv.NewWriter(&updatesBuf)
+func MakeCSVBytes(
+	records *[]sdk.Record,
+	prefix string,
+	orderingColumns []string,
+	insertsBuf *bytes.Buffer,
+	updatesBuf *bytes.Buffer,
+) (map[string]string, []string, []string, error) {
```

Comment:
Returning maps and slices will mark them as leaked variables (can be transferred from stack to heap). That could end up in unexpected allocations and data copy.

Follow the [io.Copy](https://pkg.go.dev/io#Copy) example and put the destination as part of the parameters.

Doing that will also allow you to create [memory pool](https://pkg.go.dev/sync#Pool) for those destination parameters (on who is calling the method), which will reuse those slabs of memory, avoiding allocation, improving the performance even more.

Reply:
https://github.com/conduitio-labs/conduit-connector-snowflake/pull/104/commits/407b614e42d17045834e691991b44a0e32b8adb2